plugins/formal-verification/src/org/jetbrains/kotlin/formver/conversion/StmtConversionVisitor.kt

```diff
@@ -211,4 +213,32 @@ class StmtConversionVisitor : FirVisitor<Exp, StmtConversionContext>() {
         }
         TODO("Handle other kinds of smart casts.")
     }
+
+    override fun visitBinaryLogicExpression(binaryLogicExpression: FirBinaryLogicExpression, data: StmtConversionContext): Exp {
+        val returnVar = data.newAnonVar(BooleanTypeEmbedding)
+        data.addDeclaration(returnVar.toLocalVarDecl())
+        val left = binaryLogicExpression.leftOperand.accept(this, data)
+        val rightSubStmt = StmtConverter(data)
```

Comment:
This might be a place where a comment could be helpful. I was a bit confused until I realized that you are only using the "MethodConversionContext part" of data (as data has to be of type StmtConverter, too).

Reply:
Is this case particularly different than what we do in when expressions or while loops?  I feel like if it's unclear, perhaps we need a more idiomatic way to construct a block.  (The stuff with ad-hoc return vars is probably worth cleaning up in any case, though I'm not entirely sure how best to do it.)

Reply:
Yeah you're right. Maybe a way to make it clearer is to have `StmtConversionContext` not inherit from `MethodConversionContext` but have an object of `MethodConversionContext` as a field. I think this composition would make it cleaner and distinguish the two a bit (could be put into a separate PR).